actor/v7action/service_instance.go

```diff
@@ -59,16 +60,29 @@ func (actor Actor) UpdateUserProvidedServiceInstance(serviceInstanceName, spaceG
 
 func (actor Actor) UpdateManagedServiceInstance(serviceInstanceName, spaceGUID string, serviceInstanceUpdates ServiceInstanceUpdateManagedParams) (Warnings, error) {
 	var (
-		original resources.ServiceInstance
-		jobURL   ccv3.JobURL
+		original    resources.ServiceInstance
+		jobURL      ccv3.JobURL
+		allWarnings Warnings
 	)
 
+	if serviceInstanceUpdates.ServicePlanName.IsSet {
```

Comment:
I think there are a couple of ways this could be improved:
1) Ideally we should only query plans from the same service offering as the existing service plan
2) This could be incorporated as a step in the "railway" structure below

As there are follow-on stories, I'm happy to defer improvements to them